src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -346,46 +448,105 @@ private AIState cookFood()
         }
         walkTo = null;
         setDelay(STANDARD_DELAY);
+        // TODO: why COOK_COOK_FOOD instead of getState()?
         return COOK_COOK_FOOD;
     }
 
+    /**
+     * Check if something is waiting to be cooked in at least one oven.
+     *
+     * @return true if so.
+     */
+    private boolean isSomethingInOven()
+    {
+        for (final BlockPos pos : ((BuildingCook) getOwnBuilding()).getFurnaces())
+        {
+            final TileEntity entity = world.getTileEntity(pos);
+            if (entity instanceof TileEntityFurnace && !((TileEntityFurnace) entity).isBurning()
+                && !ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(COOK_SLOT)))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+    /**
+     * getPositionOfOvenToRetrieveFrom
+     *
+     * Find the first furnace in the building which is not cooking and either has a cooked result or has an uncooked input
+     * and set the worker's status to "Retrieving" and return that furnace's position.
+     *
+     * @return location of a matching furnace or null if no furnace found
+     */
     private BlockPos getPositionOfOvenToRetrieveFrom()
     {
         for (final BlockPos pos : ((BuildingCook) getOwnBuilding()).getFurnaces())
         {
+            // TODO: why do we care if the furnace is not burning?  burning && food to cook, maybe, but not burning in general.
+            // TODO: Why do we care about food in the cook slot but not fuel? both are required.
+            // TODO: It makes more sense to find ovens which have cooked food (no matter what else is going on), or will have cooked food (either because it is in the process of
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)